tuf/api/metadata.py

```diff
@@ -494,32 +535,95 @@ def __init__(
             self, _type: str, version: int, spec_version: str,
             expires: datetime, meta: JsonDict) -> None:
         super().__init__(_type, version, spec_version, expires)
-        # TODO: Add class for meta
-        self.meta = meta
+
+        self.meta = {}
+        for meta_path, meta_info in meta.items():
+            self.meta[meta_path] = MetaFile(meta_info['version'],
+                meta_info.get('length'), meta_info.get('hashes'))
+
 
     # Serialization.
     def to_dict(self) -> JsonDict:
         """Returns the JSON-serializable dictionary representation of self. """
         json_dict = super().to_dict()
+        meta_dict = {}
+        for meta_path, meta_info in self.meta.items():
+            meta_dict[meta_path] = meta_info.to_dict()
+
         json_dict.update({
-            'meta': self.meta
+            'meta': meta_dict
         })
         return json_dict
 
 
     # Modification.
-    def update(
+    def update_meta_file(
```

Comment:
I have mixed feelings about this API change. I can understand why you wanted to make the methods' names less ambiguous but it starts reminding me of the current implementation with long function names and even longer parameters list.

Maybe when the update is on a single `MetaFile` the `update()` method should be implemented by the `MetaFile` class. This way the interface will very roughly look like `Snapshot.{rolename}.update(version, length, hashes`).
Also you are not updating the `MetaFile` object but creating a new one which may or may not be a problem.

Since the classes' design and their interaction is a WIP we can revise this later, not in this PR.
Any thoughts on this @lukpueh?

Reply:
I understand your concerns and you have a point.
I realized it's an ambiguous name when I and @jku were discussing something about the new API and he misunderstood what the update function does.

Also on that point, @lukpueh said `MetaFile` is a little ambiguous name too https://github.com/theupdateframework/tuf/pull/1223#discussion_r531060868 and I agreed, but then when I renamed the class to `MetadataFileInfo` I feel it became too long...
What do you think @sechkova @lukpueh?

Reply:
That's a different case of ambiguity 😁 
My issue is not that much about the name itself, more on the uniformity of the interface and the classes design.

Reply:
This reminds me of the conversation about the `update` methods we had in https://github.com/theupdateframework/tuf/pull/1193#discussion_r514572393 pp. Let me repurpose my comment from that discussion:

I would say the name "update" in `{Targets, Snapshot, Timestamp}.update(...)` does not stand for  *"__update__ attributes of object"*  but rather for *"update attributes of object in the course of a repository __update__"*. I am talking about the Update in The Update Framework.

Right now a TUF integrator needs to call exactly these `update` methods explicitly whenever target files are added, thus above meaning of *update* made sense to me.

A bit further down the road, I want us to have a repository controller class that provides an `update` method that handles the whole chain of updates on related targets, snapshot, timestamp objects, whenever target files are added (see #1134).

Once we implement that logic in a controller, and integrators no longer need to handle updates of individual roles explicitly, above meaning of the method names becomes a bit less important. I suggest that we then (or already now) use a bit more verbose and mnemonic names than `update`.

Regarding the concerns about the length of `MetadataFileInfo` or `update_meta_file`, I can see where they come from, but I think there are still worlds between this and something like ...

```python
def _generate_and_write_metadata(rolename, metadata_filename, targets_directory, metadata_directory, storage_backend, consistent_snapshot=False, filenames=None, allow_partially_signed=False, increment_version_number=True, repository_name='default', use_existing_fileinfo=False, use_timestamp_length=True, use_timestamp_hashes=True, use_snapshot_length=False, use_snapshot_hashes=False):
```

Reply:
> A bit further down the road, I want us to have a repository controller class that provides an update method that handles the whole chain of updates on related targets, snapshot, timestamp objects, whenever target files are added (see #1134).
Once we implement that logic in a controller, and integrators no longer need to handle updates of individual roles explicitly, above meaning of the method names becomes a bit less important. I suggest that we then (or already now) use a bit more verbose and mnemonic names than update.

I agree with this.
When we have a controller, then eventually our users will stop using those `update` methods directly and we would be able to rename them with something less ambiguous.
I think the best solution here will be to drop that commit and file an issue describing when it's suitable to make that change and discuss even if we want that change in the first place.
Agreed @lukpueh, @sechkova?

Reply:
After a discussion with Lukas, we both agreed that `MetadataInfo` sounds like the best balance between clear, but yet a not verbose name for the class used to describing `meta` for Snapshot and Timestamp.